cuqi/implicitprior/_regularizedGaussian.py

```diff
@@ -113,15 +128,50 @@ def _parse_regularization_input_arguments(self, proximal, projector, constraint,
         elif (isinstance(constraint, str) and constraint.lower() == "box"):
             lower = optional_regularization_parameters["lower_bound"]
             upper = optional_regularization_parameters["upper_bound"]
-            self._proximal = lambda z, gamma: ProjectBox(z, lower, upper)
+            self._proximal = lambda z, _: ProjectBox(z, lower, upper)
             self._preset = "box" # Not supported in Gibbs
         elif (isinstance(regularization, str) and regularization.lower() in ["l1"]):
-            strength = optional_regularization_parameters["strength"]
-            self._proximal = lambda z, gamma: ProximalL1(z, gamma*strength)
+            self._strength = optional_regularization_parameters["strength"]
+            self._proximal = lambda z, gamma: ProximalL1(z, gamma*self._strength)
             self._preset = "l1"
+        elif (isinstance(regularization, str) and regularization.lower() in ["tv"]):
+            self._strength = optional_regularization_parameters["strength"]
+            if isinstance(self.geometry, (Continuous1D)):
+                self._transformation = FirstOrderFiniteDifference(self.geometry.par_dim, bc_type='zero')
+            elif isinstance(self.geometry, (Continuous2D, Image2D)):
```

Comment:
If Image2D geometry has attribute "visual_only" as true I am not sure we want to actually use a 2D TV prior. Perhaps raise an error in this case for now.

Reply:
I have now added this condition